src/OrbitGl/TimerTrack.cpp

```diff
@@ -331,6 +331,7 @@ void TimerTrack::OnTimer(const TimerInfo& timer_info) {
 
   TextBox& text_box = timer_chain->emplace_back();
   text_box.SetTimerInfo(timer_info);
+  timer_chain->emplace_back(text_box);
```

Comment:
That doesn't make sense to me. The text_box is added two lines above: `TextBox& text_box = timer_chain->emplace_back();`

Now we are adding it twice?

Reply:
Maybe emplace_back is not working as it should?

Reply:
Also there is code similar to this in `GraphTrack` and `ThreadTrack`. Does this also fix that? I would assume not.

Reply:
My guess is that is has something to do with `min_timestamp_` and `max_timestamp_` in `TimerBlock`. They are only updated when `emplace_back` is called, but not when `text_box.SetTimerInfo` is called.

Reply:
Yes, you are right. I only realized that was working and didn't analize the code.

Reply:
> My guess is that is has something to do with min_timestamp_ and max_timestamp_

Yes.

Reply:
Nono, the fix is working, the question is why.

Reply:
emplace_back is not returning void?

text_box = timer_chain-&gt;emplace_back();

Reply:
Ignore my previous comment. I don't understand why the timers are appearing.

Reply:
It behaves like emplace_black in the standard containers, so yes, it makes sense.

But this doesn't work with preserving invariance in TimerChain

Reply:
I'm not sure how to fix this. Of course a quick fix would/should work:

```
TextBox text_box{};
text_box.SetTimerInfo(timer_info);
timer_chain->emplace_back(std::move(text_box));
```

But the underlying problem persists.

Reply:
Doesn't that kind of defeat the performance optimization that the `emplace_back` change was supposed to introduce?

A short term fix is to remove `SetTimerInfo` and use a constructor parameter.